src/reducers/pipelineReducer.js

```diff
@@ -0,0 +1,31 @@
+import { Map, List } from 'immutable';
+import { updateActivityList, pushDatasetToList } from './common';
+import {
+  FLAG_GETTING_PIPELINES,
+  FLAG_GETTING_PIPELINE,
+  GET_PIPELINE_SUCCESS,
+  GET_PIPELINES_SUCCESS,
+  FLAG_CLEARING_PIPELINE,
+  LOAD_PIPELINE_SUCCESS,
+} from '../shared/types';
+
+const INITIAL_STATE = Map({
+  activity: List(),
+  pipelines: List(),
+});
+
+export default (state = INITIAL_STATE, { type, payload }) => {
+  switch (type) {
+    case FLAG_GETTING_PIPELINES:
+    case FLAG_GETTING_PIPELINE:
+    case GET_PIPELINE_SUCCESS:
+    case FLAG_CLEARING_PIPELINE:
```

Comment:
I don't think you implemented the clearing function yet. So either implement it or delete this type.